api-reference/10 UI Widgets/dxDataGrid/1 Configuration/onEditorPreparing.md

```diff
@@ -397,5 +259,7 @@ Use the **parentType** function parameter to check if the editor being customize
     url: "https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/CommandColumnCustomization/"
 }
 
+[note]**onEditorPreparing** in not an appropriate function to set a default value for an editor. Use the [onInitNewRow]({basewidgetpath}/Configuration/#onInitNewRow) function for this purpose.
```

Comment:
"in not an appropriate function" Why is is not 'appropriate'?

Reply:
The reason is this option allows us to set an editor's value when the editor belongs to an existing row, but it is a bad idea to use it to set default value when the row (or its editor) is created in runtime. This approach will cause numerous issues and is not recommended.

Reply:
We do not recommend that you use the **onEditorPreparing** function to specify an editor's default value. Use the [onInitNewRow]({basewidgetpath}/Configuration/#onInitNewRow) function instead.